# Resolução Questão 2

Imports

In [1]:
import pandas as pd
from sqlite3 import connect

Reading and cleaning database

In [2]:
database = pd.read_excel('./Mini-case FP&A_shared.xlsx', sheet_name='Database')
database.drop('Unnamed: 0', axis=1, inplace=True)
database.drop(index=0 , axis=0, inplace=True)
database.rename(columns={'Unnamed: 1': 'school_id', 'Unnamed: 2': 'school_state', 'Unnamed: 3': 'school_students', 'Unnamed: 4': 'tuition_value'}, inplace=True)
database.reset_index(drop=True)

school_id school_state school_students tuition_value
0        7140091498           BA             952           915
1       17847374083           TO             929          3422
2          78996896           RN             533           210
3        3195632885           MG             357          3170
4       11604354948           MS            1607          3285
...             ...          ...             ...           ...
499995  10386539343           SC            1900          3298
499996   1248084019           MA            1104          4282
499997   2728548975           AP            1476          1104
499998    451021570           RS             329          1919
499999   6734063721           DF             321          1443

[500000 rows x 4 columns]

Initializing connection to database

In [3]:
conn = connect(':memory:') # Criando banco SQL
database.to_sql('database', conn)

500000

Resolution question 2 - a)

In [4]:
question2a = pd.read_sql('SELECT school_state, '
                         'count(DISTINCT school_id) AS qtt_schools '
                         'FROM database '
                         'GROUP BY 1 '
                         'ORDER BY 2 DESC',
                         conn)
question2a

school_state  qtt_schools
0            RR        18772
1            MT        18755
2            RO        18686
3            MG        18637
4            BA        18604
5            SP        18596
6            AL        18593
7            DF        18582
8            PI        18571
9            AP        18569
10           SE        18544
11           RJ        18543
12           AM        18524
13           GO        18501
14           SC        18498
15           PR        18496
16           RN        18487
17           PE        18475
18           CE        18468
19           PA        18458
20           TO        18456
21           RS        18446
22           MS        18432
23           PB        18403
24           MA        18355
25           ES        18351
26           AC        18198

R: RR, com 18772 escolas

Resolution question 2 - b)

In [5]:
question2b = pd.read_sql('SELECT school_state, sum(school_students) AS qtt_students '
                         'FROM database '
                         'GROUP BY 1 '
                         'ORDER BY 2 DESC',
                         conn)
question2b

school_state  qtt_students
0            RO      19772326
1            RR      19721405
2            SP      19720216
3            MG      19688132
4            MT      19639862
5            DF      19610657
6            AL      19580230
7            AP      19554884
8            BA      19548986
9            PI      19532306
10           AM      19518113
11           MS      19471350
12           RJ      19465055
13           PR      19461679
14           RS      19453616
15           CE      19442823
16           SE      19439804
17           GO      19420795
18           PB      19391349
19           SC      19383100
20           MA      19347955
21           RN      19343487
22           PA      19335222
23           PE      19289850
24           TO      19251462
25           ES      19199869
26           AC      19027049

R: RO, com 19.772.326 alunos

Resolution question 2 - c)

In [6]:
question2c = pd.read_sql("SELECT school_state, AVG(tuition_value) as tuition_mean FROM database WHERE school_state LIKE '%BA%' GROUP BY 1", conn)
question2c

school_state  tuition_mean
0           BA   2604.833692

R: BA, média de 2604.83 de mensalidade.

Resolution question 2 - d)

In [9]:
question2d = pd.read_sql("SELECT school_id, school_students, AVG(tuition_value) as tuition_mean FROM database WHERE (school_students >= 100 and school_students <= 400) GROUP BY school_id", conn)
question2d # Mensalidade media de cada escola

school_id  school_students  tuition_mean
0           429227              139        4566.0
1           678733              339        1043.0
2           702844              123        3166.0
3          1023719              305        2016.0
4          1044269              286        4851.0
...            ...              ...           ...
78862  19999064628              265        4638.0
78863  19999285107              191         417.0
78864  19999553373              118        2001.0
78865  19999758391              139        2974.0
78866  19999861769              331        2140.0

[78867 rows x 3 columns]

In [10]:
question2d['tuition_mean'].mean() # Mensalidade media de todas as escolas com a quantidade de alunos de 100 a 400.

2603.991834354039

R: A média das escolas que possuem entre 100 e 400 alunos é de 2603.99